Remotion/ObjectBinding/Web.ClientScript/Scripts/BocList.ts

```diff
@@ -0,0 +1,779 @@
+  
+    fakeTableHeadContainer.show();
+  }
+
+  public static FixHeaderPosition(tableContainer: JQuery, scrollableContainerJQuery: JQuery): void
+  {
+    var scrollableContainer = scrollableContainerJQuery[0];
+    var fakeTableHeadContainer = tableContainer.children('div.bocListFakeTableHead').first()[0];
+    var scrollLeft = scrollableContainer.scrollLeft;
+    var previousScrollLeft = scrollableContainerJQuery.data("bocListPreviousScrollLeft");
+    var fakeTableHeadScrollLeft = fakeTableHeadContainer.scrollLeft;
+
+    var hasScrollMoveFromScrollbar = previousScrollLeft !== scrollLeft;
+    var hasScrollMoveFromColumnHeader = previousScrollLeft !== fakeTableHeadScrollLeft;
+    if (!hasScrollMoveFromScrollbar && !hasScrollMoveFromColumnHeader)
+      return;
+
+    scrollLeft = hasScrollMoveFromColumnHeader ? fakeTableHeadScrollLeft : scrollLeft;
+    scrollableContainerJQuery.data("bocListPreviousScrollLeft", scrollLeft);
+
+    if (hasScrollMoveFromColumnHeader)
+      scrollableContainer.scrollLeft = scrollLeft;
+
+    if (hasScrollMoveFromScrollbar)
+      fakeTableHeadContainer.scrollLeft = scrollLeft;
+  }
+
+  public static InitializeNavigationBlock (pageNumberField: JQuery, pageIndexField: JQuery): void
+  {
+    ArgumentUtility.CheckNotNullAndTypeIsObject ('pageNumberField', pageNumberField);
+    ArgumentUtility.CheckNotNullAndTypeIsObject('pageIndexField', pageIndexField);
+  
+    pageNumberField.bind('change', function () {
+  
+      var pageNumber = parseInt(pageNumberField.val(), 10);
+      if (isNaN (pageNumber) || !TypeUtility.IsInteger (pageNumber))
+      {
+        if (pageNumberField.val().length > 0)
+          setTimeout(function () { pageNumberField.focus(); }, 0);
+  
+        pageNumberField.val(String(parseInt(pageIndexField.val(), 10) + 1));
+        return false;
+      }
+      else
+      {
+        pageIndexField.val (String(pageNumber - 1));
+        pageIndexField.trigger('change');
+        return true;
+      }
+    });
+  
+    pageNumberField.bind('keydown', function (event) {
+      var enterKey = 13;
+      var zeroKey = 48;
+      var nineKey = 57;
+      var zeroKeyNumBlock = 96;
+      var nineKeyNumBlock = 105;
+      var f1Key = 112;
+      var f12Key = 123;
+      var isEnterKey = event.keyCode == enterKey;
+      var isControlKey = event.keyCode < zeroKey || event.keyCode >= f1Key && event.keyCode <= f12Key;
+      var isNumericKey = event.keyCode >= zeroKey && event.keyCode <= nineKey || event.keyCode >= zeroKeyNumBlock && event.keyCode <= nineKeyNumBlock;
+  
+      if (isEnterKey)
+      {
+        pageNumberField.trigger("change");
+        event.cancelBubble = true;
+        event.stopPropagation();
+        return false;
+      }
+      else if (event.altKey || event.ctrlKey || isControlKey || isNumericKey)
+      {
+        return true;
+      }
+      else
+      {
+        return false;
+      }
+    });
+  }
+}
+
+function BocList_InitializeGlobals(): void
```

Comment:
Check for each global function if they are used from C# code or not. If they are not we can remove them from here and make the class member `private`.